CommunityToolkit.App.Shared/Renderers/ToolkitSampleRenderer.xaml

```diff
@@ -211,25 +210,41 @@
                     <Pivot x:Name="CodePivot"
                            MaxHeight="400"
                            Margin="16,0,16,0"
-                           HorizontalAlignment="Stretch">
+                           HorizontalAlignment="Stretch"
+                           HorizontalContentAlignment="Stretch">
                         <Pivot.Resources>
                             <x:Double x:Key="PivotHeaderItemFontSize">14</x:Double>
                         </Pivot.Resources>
 
                         <PivotItem Header="XAML">
                             <ScrollViewer>
-                                <TextBlock wasm:IsTextSelectionEnabled="True"
-                                           win:IsTextSelectionEnabled="True"
-                                           Style="{StaticResource CaptionTextBlockStyle}"
-                                           Text="{x:Bind XamlCode, Mode=OneWay}" />
+                                <StackPanel Orientation="Vertical">
+                                    <xamarin:TextBlock x:Name="XAMLCodeRenderer"
+                                                       Margin="0,12,0,12"
+                                                       wasm:IsTextSelectionEnabled="True"
+                                                       FontFamily="Consolas"
+                                                       FontSize="12"
+                                                       TextWrapping="WrapWholeWords" />
+                                    <win:RichTextBlock x:Name="XAMLCodeRenderer"
+                                                       FontFamily="Consolas"
+                                                       FontSize="12"
+                                                       IsTextSelectionEnabled="True" />
+                                </StackPanel>
                             </ScrollViewer>
                         </PivotItem>
                         <PivotItem Header="C#">
                             <ScrollViewer>
-                                <TextBlock wasm:IsTextSelectionEnabled="True"
-                                           win:IsTextSelectionEnabled="True"
-                                           Style="{StaticResource CaptionTextBlockStyle}"
-                                           Text="{x:Bind CSharpCode, Mode=OneWay}" />
+                                <StackPanel Orientation="Vertical">
+                                    <xamarin:TextBlock x:Name="CSharpCodeRenderer"
```

Comment:
As an add-on to this, we could take the same approach I did for the `MarkdownTextBlock` in the sample app, and create a wrapper around TextBlock, but on WASM we call the ColorCode  HTML generator and then inject that HTML into the control...

Reply:
Actually you could just encapsulate all this XAML stuff and the code into the control, like the MarkdownTextBlock helper does: https://github.com/CommunityToolkit/Tooling-Windows-Submodule/blob/main/CommunityToolkit.App.Shared/Renderers/Markdown/MarkdownTextBlock.cs

Reply:
Ugh, forgot about that (and thus why sealing UI controls is generally annoying...) - Linking to https://github.com/microsoft/microsoft-ui-xaml/issues/780 as usual...

I guess for the native platform you could have a `ContentPresenter` or something and stick in a RTB, or create a local UserControl or something that just wraps a RTB within it? Have it expose a `Text` property?

Reply:
Was thinking maybe we can just use composition instead of inheritance, i.e. wrap everything in just a `Control`, and that control can setup it's content based on platform, on WinUI2/3 it sets it as a RTB and uses ColorCode directly, for WASM it sets it to a TextBlock or whatever but replaces the HTML content with the ColorCode output, and then on other Uno targets it just uses a TextBlock with the raw text? The control can just expose a general `Text` property that can be passed in the Markdown from the ToolkitSampleRenderer, so then the renderer doesn't care about the platform.

@niels9001 what did you want to do here? Should we just commit this as-is for now or did you want to take a stab at the above?